### PR TITLE
Change EDGEDB_TLS_VERIFY_HOSTNAME to EDGEDB_CLIENT_TLS_SECURITY

### PR DESCRIPTION
**Pull request by fmoor**
implements https://github.com/edgedb/edgedb/discussions/3049#discussioncomment-1507123

### CODE REVIEW

File: connection_testcases.json [hunk 1]
```diff
@@ -1383,7 +1431,7 @@
   {
     "env": {
       "EDGEDB_DSN": "edgedb://?tls_verify_hostname=yes",
```
[the remark this answers is not in the document]
Comment (jaclarke) [reply connection_testcases.json:1467]: So to clarify: the `createClient()` config option is now named `tlsSecurity` (or `tls_security`), the env var is `EDGEDB_CLIENT_TLS_SECURITY`, in DSN's it's `tls_security`, and credentials file is the only place we still allow both `tls_verify_hostname` and `tls_security` (as long as they agree). Is this right?

[the remark this answers is not in the document]
Comment (elprans) [reply connection_testcases.json:1467]: Seems right.

File: connection_testcases.json [hunk 2]
```diff
@@ -1401,153 +1435,98 @@
...
-    },
     "env": {
-      "EDGEDB_DSN": "edgedb://",
-      "EDGEDB_TLS_VERIFY_HOSTNAME": "true"
-    },
-    "fs": {
-      "files": {
-        "/home/edgedb/test/cert.pem": "-----BEGIN CERTIFICATE-----\nMIICEjCCAXsCAg36MA0GCSqGSIb3DQEBBQUAMIGbMQswCQYDVQQGEwJKUDEOMAwG\nA1UECBMFVG9reW8xEDAOBgNVBAcTB0NodW8ta3UxETAPBgNVBAoTCEZyYW5rNERE\nMRgwFgYDVQQLEw9XZWJDZXJ0IFN1cHBvcnQxGDAWBgNVBAMTD0ZyYW5rNEREIFdl\nYiBDQTEjMCEGCSqGSIb3DQEJARYUc3VwcG9ydEBmcmFuazRkZC5jb20wHhcNMTIw\nODIyMDUyNjU0WhcNMTcwODIxMDUyNjU0WjBKMQswCQYDVQQGEwJKUDEOMAwGA1UE\nCAwFVG9reW8xETAPBgNVBAoMCEZyYW5rNEREMRgwFgYDVQQDDA93d3cuZXhhbXBs\nZS5jb20wXDANBgkqhkiG9w0BAQEFAANLADBIAkEAm/xmkHmEQrurE/0re/jeFRLl\n8ZPjBop7uLHhnia7lQG/5zDtZIUC3RVpqDSwBuw/NTweGyuP+o8AG98HxqxTBwID\nAQABMA0GCSqGSIb3DQEBBQUAA4GBABS2TLuBeTPmcaTaUW/LCB2NYOy8GMdzR1mx\n8iBIu2H6/E2tiY3RIevV2OW61qY2/XRQg7YPxx3ffeUugX9F4J/iPnnu1zAxxyBy\n2VguKv4SWjRFoRkIfIlHX0qVviMhSlNy2ioFLy7JcPZb+v3ftDGywUqcBiVDoea0\nHn+GmxZA\n-----END CERTIFICATE-----\n"
-      }
+      "EDGEDB_DSN": "edgedb://?tls_verify_hostname=yes",
```
Comment (jaclarke) [inline connection_testcases.json]: This dsn is still using `tls_verify_hostname`

File: connection_testcases.json [hunk 3]
```diff
@@ -1401,153 +1435,98 @@
...
-    },
     "env": {
-      "EDGEDB_DSN": "edgedb://",
-      "EDGEDB_TLS_VERIFY_HOSTNAME": "yes"
-    },
-    "fs": {
-      "files": {
-        "/home/edgedb/test/cert.pem": "-----BEGIN CERTIFICATE-----\nMIICEjCCAXsCAg36MA0GCSqGSIb3DQEBBQUAMIGbMQswCQYDVQQGEwJKUDEOMAwG\nA1UECBMFVG9reW8xEDAOBgNVBAcTB0NodW8ta3UxETAPBgNVBAoTCEZyYW5rNERE\nMRgwFgYDVQQLEw9XZWJDZXJ0IFN1cHBvcnQxGDAWBgNVBAMTD0ZyYW5rNEREIFdl\nYiBDQTEjMCEGCSqGSIb3DQEJARYUc3VwcG9ydEBmcmFuazRkZC5jb20wHhcNMTIw\nODIyMDUyNjU0WhcNMTcwODIxMDUyNjU0WjBKMQswCQYDVQQGEwJKUDEOMAwGA1UE\nCAwFVG9reW8xETAPBgNVBAoMCEZyYW5rNEREMRgwFgYDVQQDDA93d3cuZXhhbXBs\nZS5jb20wXDANBgkqhkiG9w0BAQEFAANLADBIAkEAm/xmkHmEQrurE/0re/jeFRLl\n8ZPjBop7uLHhnia7lQG/5zDtZIUC3RVpqDSwBuw/NTweGyuP+o8AG98HxqxTBwID\nAQABMA0GCSqGSIb3DQEBBQUAA4GBABS2TLuBeTPmcaTaUW/LCB2NYOy8GMdzR1mx\n8iBIu2H6/E2tiY3RIevV2OW61qY2/XRQg7YPxx3ffeUugX9F4J/iPnnu1zAxxyBy\n2VguKv4SWjRFoRkIfIlHX0qVviMhSlNy2ioFLy7JcPZb+v3ftDGywUqcBiVDoea0\nHn+GmxZA\n-----END CERTIFICATE-----\n"
-      }
+      "EDGEDB_DSN": "edgedb://?tls_verify_hostname=yes",
```
Comment (jaclarke) [inline connection_testcases.json]: ...and this dsn

File: connection_testcases.json [hunk 4]
```diff
@@ -2047,8 +1974,120 @@
...
+      "dsn": "test_instance"
+    },
+    "fs": {
+      "homedir": "/home/edgedb",
+      "files": {
+        "/home/edgedb/.config/edgedb/credentials/test_instance.json": "{\"port\": 10702, \"user\": \"edgedb\", \"tls_verify_hostname\": true, \"tls_security\": \"insecure\"}"
+      }
+    },
+    "error": {
+      "type":"invalid_tls_security"
```
Comment (jaclarke) [inline connection_testcases.json:2038]: I think this should be an `invalid_credentials_file` error?

File: connection_testcases.json (hunk 2 above)
Comment (fmoor) [reply connection_testcases.json]: I think this diff is out of date :confused:

**Review by jaclarke — APPROVED** [review]

**Pull request merged**